crates/cac_client/src/lib.rs

```diff
@@ -204,11 +238,33 @@ impl Client {
         filter_keys: Option<Vec<String>>,
         merge_strategy: MergeStrategy,
     ) -> Result<Map<String, Value>, String> {
-        let mut cac = self.eval(query_data, merge_strategy).await?;
-        if let Some(keys) = filter_keys {
-            cac = filter_keys_by_prefix(cac, keys);
-        }
-        Ok(cac)
+        let filter_string = if let Some(mut vec) = filter_keys.clone() {
+            vec.sort();
+            vec.join(",")
+        } else {
+            "null".to_string()
+        };
+        let hash_key = json_to_sorted_string(&json!(query_data))
+            + "$"
```

Comment:
can we use different delimiter here and the next one, it can cause collision as `json_to_sorted_string` also adds ["$"](https://github.com/juspay/superposition/pull/268/files#diff-0cce34f446e173a9f372d1cc5b7a96287a40a48b0871aab96c3818d11b7fc4fbR14)

Reply:
we can use `?`